models/_sector/dex/trades/base/platforms/horizondex_base_base_trades.sql

```diff
@@ -0,0 +1,56 @@
+{{
+    config(
+    schema = 'horizondex_base'
+    , alias = 'base_trades'
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , incremental_strategy = 'merge'
+    , unique_key = ['tx_hash', 'evt_index']
+    , incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+    )
+}}
+
+WITH dexs AS
+(
+    SELECT
+        t.evt_block_number AS block_number
+        , t.evt_block_time AS block_time
+        , t.recipient AS taker
+        , cast(null as varbinary) as maker
+        , CASE WHEN t.deltaQty0 < INT256 '0' THEN abs(t.deltaQty0) ELSE abs(t.deltaQty1) END AS token_bought_amount_raw -- when amount0 is negative it means trader_a is buying token0 from the pool
+        , CASE WHEN t.deltaQty0 < INT256 '0' THEN abs(t.deltaQty1) ELSE abs(t.deltaQty0) END AS token_sold_amount_raw
+        , CASE WHEN t.deltaQty0 < INT256 '0' THEN f.token0 ELSE f.token1 END AS token_bought_address
+        , CASE WHEN t.deltaQty0 < INT256 '0' THEN f.token1 ELSE f.token0 END AS token_sold_address
```

Comment:
Hi @tomfutago,
this part seems to be very similar to [`kyberswap_compatible_trades`](https://github.com/Hosuke/spellbook/blob/main/macros/models/_sector/dex/kyberswap_compatible_trades.sql#L17), but with the tokens flipped.

Reply:
@Hosuke if we have any doubt that the token bought/sold are flipped in any given new dex, we should force `dex.trades_beta` to be modified in PR and have that run. that way we see `amount_usd` populated and then the test on that field will run to ensure no values out of proportion due to incorrect assignment. when merging new dex PRs, `dex.trades_beta` runs in prod to refresh regardless, so modifying to run is no big deal

Reply:
lets go ahead and try that here before merge